### PR TITLE
spring native-memory config support added

### PR DESCRIPTION
**Pull request by bilalyasar**
this PR solves #4772 
an example for configuring `native-memory`
```
           <hz:native-memory enabled="false" allocator-type="POOLED" metadata-space-percentage="10.2" min-block-size="10"
                              page-size="20">
                <hz:size unit="MEGABYTES" value="256"/>
           </hz:native-memory>
```
i can make <hz:size> as attribute but at that time it will not very compatible with server configuration.

### CODE REVIEW

**Comment by mesutcelik** [comment]
checkstyle failure.
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-pr-builder/12700/console

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/clientNetworkConfig-applicationContext.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
        xsi:schemaLocation="http://www.springframework.org/schema/beans
 		http://www.springframework.org/schema/beans/spring-beans-2.5.xsd
 		http://www.hazelcast.com/schema/spring
-		http://www.hazelcast.com/schema/spring/hazelcast-spring-3.2.xsd">
+		http://www.hazelcast.com/schema/spring/hazelcast-spring-3.5.xsd">
```
Comment (mesutcelik) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/clientNetworkConfig-applicationContext.xml]: you can completely remove version number and use versionless schema definition i.e. http://www.hazelcast.com/schema/spring/hazelcast-spring.xsd

**Comment by mesutcelik** [comment]
one minor comment. 
:+1:

**Comment by mesutcelik** [comment]
@serkan-ozal , 
Can you please have a look?

**Comment by bilalyasar** [comment]
verify

**Comment by bwzhang2011** [comment]
@bilalyasar, native memory only enabled and worked under enterprise hazelcast, is that right ?

**Comment by bilalyasar** [comment]
@bwzhang2011 yes, right.
i talked with @serkan-ozal  according to his advice i made very little change.

**Comment by mesutcelik** [comment]
@bilalyasar , can you please create backport of this PR?

**Pull request merged**